### PR TITLE
Reviving the revival of revivers reviving. Revival edition.

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do
Tweaked from: https://github.com/tgstation/tgstation/pull/78377
Organs now have an on_death proc that runs when the owner is dead like on_life. This is used by egg organs, and the reviver.
The reviver now heals you better, before going on cooldown depending on how much damage is healed.
Reviver tweaks:
Better healing in crit.
Will heal you when dead, and tries to revive you. The defib attempt will cause some organ damage and cost some blood.
However, will take a long time to heal depending on damage, is noticable, and will usually fail in cases such as:
Heavy blood loss
Vampirism.
Chasms
Legions
Any simple mob that sticks around since it will maul you when you revive
slaughter demons
Xenos since no heart
Terrors in most cases due to no blood
Being left in lava
Left in a plasmafire
Left in space without a suit
Changelings absorbing you
Someone poisoning you so long as it doesnt run out beforeyou revive
Anyone paying attention that doesnt want you to revive.

It’s good for accidental deaths, or run and guns where an antagonist just tries to kill you and leaves, vs actively leaves you dead. Antagonists can totally use this, but officers should catch on.
It doesnt help changelings much, outside of crit healing, as their fake revive is better in any way.
Doesnt break you out of cuffs when reviving, so easy to deal with still.
Anyone can delay it by 10 seconds by toolboxing, 10 seconds by batoning, 20 by lasering, 100 by point blank lasering in the head, etc.
Afterwords, once you are healed, the reviver will go on cooldown, depending on how much it has healed you, with a big delay of 10 minutes for each (successful) defib attempt.
Also works on ipcs now.
Overall, a useful implant for those that know they will die, as it will make them possibly get back up, or at least make their body easier to revive. Even if it can’t get you up, all your brute and burn damage being gone by the time the paramed reaches you is still useful!
Ups the tech level to bio 7 since it is more valueable.

Upgraded cybernetic heart might actually cure some shock now. As well, if it does restart your heart, it will heal some oxy damage and clear breathloss, so there is a small chance you might survive.

The hope trinket you get 50% of the time from killing a sentient pandora, now heals much better, so it is actually more useful than a reviver (in healing, doesnt revive you), since it is a hard fight.

## Why It's Good For The Game
See above. This should make the reviver much more useful, and actually viable. The healing is now much more consistant, and the reviving from death / healing dead bodies can be a fun gimic, but also useful for more chaotic sec rounds.

The upgraded cybernetic heart actually doing its medical related stuff is good.

Hope being worth the fight now is good, actually can compete with the regen gene, or a healing virus, both of which are much easier to get.

## Testing

Confirmed xenomorph eggs worked.
Confirmed reviver worked, didnt heal too fast, did visuals, didn't revive people too early, didn't revive people wiht over 180 clone damage.
Confirmed it worked on ipcs.
Confirmed hope healed at proper values.
Confirmed reviver did organ damage and blood loss correctly

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

image

## Changelog

:cl: [SSensum](https://github.com/SSensum), Qwertytoforty
tweak: Revivers now heal people when dead, and attempt to revive them from the dead. Emps, or just bashing the dead body, will delay this.
tweak: Reviver now has a higher tech level.
tweak: Upgraded cybernetic heart should be better at healing stage 2 crit and stopped hearts.
tweak: Hope should heal you much better than before.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_life.dm [hunk 1]
```diff
@@ -205,6 +207,14 @@
 		var/datum/organ/datum_organ_var_name_idk = internal_organ_datums[organ_tag]
 		datum_organ_var_name_idk.on_life()
 
+/mob/living/carbon/proc/handle_dead_organs()
+	for(var/thing in internal_organs)
+		var/obj/item/organ/internal/O = thing
+		O.on_death()
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/carbon_life.dm:213]: this shouldnt be called `on_death`, as it implies it called once when the person dies

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/carbon_life.dm (hunk 1 above)
Comment (Qwertytoforty) [reply code/modules/mob/living/carbon/carbon_life.dm:213]: Fair, was mirroring it to on_life

Comment (Qwertytoforty) [reply code/modules/mob/living/carbon/carbon_life.dm]: renamed

**Review by Burzah — APPROVED** [review]

File: code/modules/surgery/organs/augments_internal.dm [hunk 2]
```diff
@@ -663,14 +663,21 @@
 
 /obj/item/organ/internal/cyberimp/chest/reviver
 	name = "Reviver implant"
-	desc = "This implant will attempt to heal you out of critical condition. For the faint of heart!"
+	desc = "This implant will attempt to revive and heal you out of crit or death. For the faint of heart!"
```
Comment (DGamerL) [inline code/modules/surgery/organs/augments_internal.dm:666]:
```suggestion
	desc = "This implant will attempt to revive and heal you out of critical condition or death. For the faint of heart!"
```

**Review by DGamerL — APPROVED** [review]

**Pull request merged**